araproc/analysis/standard_reco.py

```diff
@@ -924,3 +926,364 @@ def get_avg_corr_snr(self, wave_packet, excluded_channels = []):
+        reco_results : dict
+            Reco results already with the specific reconstruction reqeusted (so
+            the keys of this object include 'theta' and 'phi')
+        excluded_channels : set
+            Any channels to exclude that aren't already excluded by 
+            livetime configurations.
+        which_distance : str
+            The distance of the reconstruction for analysis, following the 
+            convention of other functions in this class. 
+
+        Returns
+        -------
+        csw : ROOT.TGraph
+            The coherently summed waveform. Only returned if `return_csw==True`
+        warning : int
+            If non-zero, contains information about something that went wrong
+              in the CSW calculation.
+        """
+
+        # Initialize warning object as 0
+        warning = 0
+
+        # Check that reco_results have been passed in properly
+        if 'theta' not in reco_results.keys():
+            raise ValueError(
+                "reco_result object passed into this function does not appear "
+                "to be for a specific reconstruction. Should have a 'theta' key "
+                "but only has the following:", reco_results.keys())
+
+        # Make the excluded_channels object a set to speed up computation
+        excluded_channels = set(excluded_channels)
+
+        # Determine which channels should be used in the csw
+        channels_to_csw = []
+        for ch_ID in wavepacket['waveforms'].keys(): 
+            if ch_ID not in excluded_channels:
+                channels_to_csw.append(ch_ID)
+
+        # Determine the "reference channel" to base the CSW around as the
+        #   channel with the maximum voltage
+        reference_ch = -123456
+        reference_ch_max_voltage = -1
+        for ch_ID in channels_to_csw:
+            this_max_voltage = np.max(wavepacket['waveforms'][ch_ID].GetY())
+            if this_max_voltage > reference_ch_max_voltage:
+                reference_ch_max_voltage = this_max_voltage
+                reference_ch = ch_ID 
+
+        # Get arrival delays relative to the reference channel based on
+        #   expected arrival times from reconstruction results
+        arrival_delays_reco = self.get_arrival_delays_reco(
+            reco_results, wavepacket['waveforms'].keys(), excluded_channels, 
+            reference_ch, which_distance, solution)
+        
+        # Get arrival delays by zooming in on the cross correlation between
+        #   each channel and the reference channel around the expected
+        #   arrival delay from reconstruction results
+        arrival_delays = self.get_arrival_delays_AraRoot_xcorr(
+            wavepacket, excluded_channels, reference_ch, 
+            arrival_delays_reco, is_software)  
+        
+        # Calculate expected signal time as when the reference channel saw 
+        #   its maximum signal
+        expected_signal_time = wavepacket['waveforms'][reference_ch].GetX()[
```

Comment:
Can we wrap this `GetX()` in `np.asarray` just in case? If it does somehow create a leak it'll be a pain to find haha. So better to be safe

Reply:
Sure! Anything with .GetN should be fine right?

Reply:
Roger that thanks. It's been fixed!